### DISCUSSION TITLE
How to filter out objects from canvas.relativePan()?

### DISCUSSION THREAD

**Discussion by enriqueerrando** [fabricjs/fabric.js#9603] [Q&A]
Hi,

I have a FabricJS (latest version) canvas with the following behaviour:

- Several selectable objects added
- One rectangle that is used as an overlay semitransparent layer with selectable set to false to ensure can't be selected and it always staysin the same place
- That overlay rect has a circle clipPathed to it and with inverted set to true and so it makes a hole in the overly
- Panning using mouse events: down/move/up and I use canvas.relativePan(delta) at mouse:move
- Buttons to increase/decrease canvas zoom

Is it possible to filter out the overlay rectangle from canvas.relativePan(delta)?

Thanks in advance!

**Comment by enriqueerrando** [comment]
I've just found a solution!!

Instead of using canvas.relativePan(delta), during muse:move I filter out the overlay rectangle by looping through canvas objects with canvas.getObjects().forEach and just reposition every object apart from the overlay one.

Do u have a better solution?

Thanks in advance!

**Reply by ShaMan123** [reply]
Sounds like you are looking from `ignoreVpt: true` option of overlay
